[Bug]: `TestCheckControllerSuite/TestBasic` unit test  is not stable

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 5be9099
- Deployment mode(standalone or cluster): both
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```


### Current Behavior

`TestCheckControllerSuite/TestBasic` may fail randomly

### Expected Behavior

Unit test shall passes stably

### Steps To Reproduce

_No response_

### Milvus Log

```
2024-01-06T03:11:27.7874602Z     mock_scheduler.go:433: PASS:	Add(string)
2024-01-06T03:11:27.7875754Z     mock_scheduler.go:433: PASS:	GetSegmentTaskNum()
2024-01-06T03:11:27.7876604Z     mock_scheduler.go:433: PASS:	GetChannelTaskNum()
2024-01-06T03:11:27.7877894Z     mock_balancer.go:179: FAIL:	AssignSegment(string,string,string)
2024-01-06T03:11:27.7880612Z         		at: [/go/src/github.com/milvus-io/milvus/internal/querycoordv2/balance/mock_balancer.go:94 /go/src/github.com/milvus-io/milvus/internal/querycoordv2/checkers/controller_test.go:127]
2024-01-06T03:11:27.7883161Z     mock_balancer.go:179: PASS:	AssignChannel(string,string)
2024-01-06T03:11:27.7884588Z     mock_balancer.go:179: FAIL: 1 out of 2 expectation(s) were met.
2024-01-06T03:11:27.7885508Z         	The code you are testing needs to make 1 more call(s).
2024-01-06T03:11:27.7888607Z         	at: [/go/src/github.com/milvus-io/milvus/internal/querycoordv2/balance/mock_balancer.go:179 /usr/local/go/src/testing/testing.go:1150 /usr/local/go/src/testing/testing.go:1328 /usr/local/go/src/testing/testing.go:1570]
2024-01-06T03:11:27.7891061Z     mock_broker.go:474: PASS:	GetRecoveryInfoV2(string,int64)
2024-01-06T03:11:27.7892030Z --- FAIL: TestCheckControllerSuite (1.03s)
2024-01-06T03:11:27.7892998Z     --- FAIL: TestCheckControllerSuite/TestBasic (1.03s)
```

### Anything else?

_No response_

--- a/internal/querycoordv2/checkers/controller_test.go
+++ b/internal/querycoordv2/checkers/controller_test.go
@@ -124,15 +124,23 @@ func (suite *CheckerControllerSuite) TestBasic() {
 	suite.scheduler.EXPECT().GetSegmentTaskNum().Return(0).Maybe()
 	suite.scheduler.EXPECT().GetChannelTaskNum().Return(0).Maybe()
 
-	suite.balancer.EXPECT().AssignSegment(mock.Anything, mock.Anything, mock.Anything).Return(nil)
-	suite.balancer.EXPECT().AssignChannel(mock.Anything, mock.Anything).Return(nil)
+	assignSegCounter := atomic.NewInt32(0)
+	assingChanCounter := atomic.NewInt32(0)
+	suite.balancer.EXPECT().AssignSegment(mock.Anything, mock.Anything, mock.Anything).RunAndReturn(func(i1 int64, s []*meta.Segment, i2 []int64) []balance.SegmentAssignPlan {
+		assignSegCounter.Inc()
+		return nil
+	})
+	suite.balancer.EXPECT().AssignChannel(mock.Anything, mock.Anything).RunAndReturn(func(dc []*meta.DmChannel, i []int64) []balance.ChannelAssignPlan {
+		assingChanCounter.Inc()
+		return nil
+	})
 	suite.controller.Start()
 	defer suite.controller.Stop()
 
 	suite.Eventually(func() bool {
 		suite.controller.Check()
-		return counter.Load() > 0
-	}, 5*time.Second, 1*time.Second)
+		return counter.Load() > 0 && assignSegCounter.Load() > 0 && assingChanCounter.Load() > 0
+	}, 5*time.Second, 1*time.Millisecond)
 }
 
 func TestCheckControllerSuite(t *testing.T) {

